File: Mar 24 881. Boats to Save People (Medium)/Solution.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        // Sort people
        sort(people.begin(), people.end());
        // Init parameters
        int left = 0, right = people.size() - 1;
        int res = 0;
        // Two Pointer iteration
        while(left <= right){
            // If we can fit two people in same boat
            if(people[left] + people[right] <= limit){
                left++;
            }
            res++;
            right--;
        }
        return res;
    }
};

File: Mar 24 881. Boats to Save People (Medium)/Solution.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        # Sort people
        people.sort()
        # Init parameters
        res = 0
        left, right = 0, len(people) - 1
        # Two Pointer iteration
        while left <= right:
            # If we can fit two people in same boat
            if people[right] + people[left] <= limit:
                left += 1
            res += 1
            right -= 1
        return res
